dot/parachain/pvf/worker.go

```diff
@@ -0,0 +1,130 @@
+package pvf
+
+import (
+	"sync"
+
+	parachainruntime "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+	parachaintypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+)
+
+type worker struct {
+	workerID parachaintypes.ValidationCodeHash
+	instance *parachainruntime.Instance
+	queue    chan *workerTask
+}
+
+type workerTask struct {
+	work             parachainruntime.ValidationParameters
+	maxPoVSize       uint32
+	candidateReceipt *parachaintypes.CandidateReceipt
+	ResultCh         chan<- *ValidationTaskResult
+}
+
+func newWorker(validationCode parachaintypes.ValidationCode, queue chan *workerTask) (*worker, error) {
+	validationRuntime, err := parachainruntime.SetupVM(validationCode)
+
+	if err != nil {
+		return nil, err
+	}
+	return &worker{
+		workerID: validationCode.Hash(),
+		instance: validationRuntime,
+		queue:    queue,
+	}, nil
+}
+
+func (w *worker) run(queue chan *workerTask, wg *sync.WaitGroup) {
+	defer func() {
+		logger.Debugf("[STOPPED] worker %x", w.workerID)
+		wg.Done()
+	}()
+
+	for task := range queue {
+		w.executeRequest(task)
+	}
```

Comment:
You can run all executeRequest concurrently as a goroutine. Right now they are running one by one.